src/algebra/squarefree.lean

```diff
@@ -85,6 +85,42 @@ end
 
 end multiplicity
 
+lemma eq_zero_or_no_irreducibles_and_squarefree_iff_irreducible_sq_not_dvd
+  (r : R) [comm_monoid_with_zero R] [wf_dvd_monoid R] :
+  (r = 0 ∧ ∀ x : R, ¬irreducible x) ∨ squarefree r ↔ ∀ x : R, irreducible x → ¬ x * x ∣ r :=
```

Comment:
I never know the precedence between `∨` and `↔`. Should I read this as
```suggestion
  (r = 0 ∧ ∀ x : R, ¬irreducible x) ∨ (squarefree r ↔ ∀ x : R, irreducible x → ¬ x * x ∣ r) :=
```
If so, I wouldn't mind including those parens for clarity.

Reply:
This simplifies for `[nontrivial R]`, right? Then you just have the RHS of the `∨`. I think that might be a worthwhile special case.

Reply:
> I never know the precedence between `∨` and `↔`. Should I read this as
> 
> If so, I wouldn't mind including those parens for clarity.

No, the parens are the other way around - `∨` gets bound more tightly. I can still add those parens if you prefer?

Reply:
> This simplifies for `[nontrivial R]`, right? Then you just have the RHS of the `∨`. I think that might be a worthwhile special case.

I agree that would be a worthwhile special case, but I don't think it's true! And this was actually why I messaged you privately the other day - your counterexample of a nontrivial field breaks it. Though I think this is because of the parens issue?

Reply:
Please add the parens! It got me all confused again. :sweat_smile:

Reply:
Done, thank you!